kaldb/src/main/proto/metadata.proto

```diff
@@ -147,3 +147,29 @@ message DatasetPartitionMetadata {
   // List of Kafka partition ids
   repeated string partitions = 3;
 }
+
+// Schema definitions.
+message LuceneFieldDef {
+  // Name of the field
+  string name = 1;
+
+  // Type of the field
+  string type = 2;
+
+  // Store the value as a Lucene StoredField.
+  bool isStored = 3;
+  // Index the field in Lucene based on it's type for efficient point queries.
+  bool isIndexed = 4;
+  // Store doc values for a field so it's more efficient for aggregations.
+  bool storeDocValue = 5;
+}
+
+// This proto documents the schema for a chunk.
+message ChunkSchema {
+  // Name of the chunk.
+  string name = 1;
+  // Field map for each field in the lucene index in the chunk.
+  map<string, LuceneFieldDef> fieldDefMap = 2;
```

Comment:
This probably shouldn't be suffixed map, since none of the other maps are suffixed map - even in the same class.

Reply:
Tried removing the map, but the code is less readable. This is a map from an app perspective for field definitions and not meant to reflect the type (which also happens to match). So leaving it as is.